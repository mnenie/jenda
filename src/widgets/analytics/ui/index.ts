import UsersChart from './UsersChart.vue';

export { UsersChart };
