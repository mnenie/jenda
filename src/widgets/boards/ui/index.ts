import TeamBoards from './TeamBoards.vue';
import ActivityСharts from './ActivityСharts.vue';

export { TeamBoards, ActivityСharts };
