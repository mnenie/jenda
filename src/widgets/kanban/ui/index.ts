import KanbanWrapper from './KanbanWrapper.vue';

export { KanbanWrapper };
