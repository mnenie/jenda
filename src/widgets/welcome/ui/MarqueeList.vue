<script setup lang="ts">
const stack = [
  '–ú—ã –∑–Ω–∞–µ–º',
  '–í—ã –Ω–∞—Å–ª–∞–∂–¥–∞–µ—Ç–µ—Å—å',
  'üëë –õ—É—á—à–∏–π –∞–Ω–∞–ª–æ–≥ –≤ –†–§',
  'Open source',
  'üëª –£ –Ω–∞—Å –ø—Ä–æ—â–µ',
  '–ë–µ–∑ —Ä–µ–∫–ª–∞–º—ã',
  '–°–æ–≤–µ—Ä—à–µ–Ω—Å—Ç–≤–æ —Ä—è–¥–æ–º ü§ü'
] as const;
</script>

<template>
  <div :class="$style.container">
    <Vue3Marquee>
      <div v-for="(s, index) in stack" :key="index" :class="$style.items">
        <p class="text-lg">{{ s }}</p>
      </div>
    </Vue3Marquee>
  </div>
</template>

<style module lang="scss">
.container {
  margin: 80px 0 150px 0;
  width: 100%;
  height: 60px;

  .items {
    display: flex;
    align-items: center;
    margin-right: 60px;
  }

  & > p {
    font-weight: 600 !important;
    line-height: 1.25;
  }
}
</style>
