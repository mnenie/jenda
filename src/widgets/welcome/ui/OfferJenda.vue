<script setup lang="ts">
import { X } from 'lucide-vue-next';
import { ref } from 'vue';

const open = ref(true);

function onCloseOffer() {
  open.value = !open.value;
}
</script>

<template>
  <div v-if="open" :class="$style.container">
    <div :class="$style.content">
      <p class="text-xs">Следите за нашими новостями в
        <a href="https://github.com/mnenie/jenda" rel="noreferrer">github</a>
      </p>
    </div>
    <X :size="13" :class="$style.icon" @click="onCloseOffer" />
  </div>
</template>

<style module lang="scss">
@import '@/app/styles/mixins';
.container {
  position: relative;
  background-color: var(--zinc-800);
  color: var(--zinc-200);
  border-radius: 10px;
  padding: 10px 20px;
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;

  .content {
    display: inline-flex;
    gap: 10px;
    align-items: center;

    & a {
      text-decoration: underline;
    }
  }

  .icon{
    position: absolute;
    top: 50%;
    color: var(--zinc-300);
    transform: translateY(-50%);
    right: 20px;
    cursor: pointer;
    @include transition;

    &:hover{
      color: var(--zinc-200);
    }
  }
}
</style>
