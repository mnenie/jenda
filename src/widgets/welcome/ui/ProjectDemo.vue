<script setup lang="ts">
import { useDark } from '@vueuse/core';
import { ProjectGrid } from '@/shared/assets/icons';
import { computed } from 'vue';

const isDark = useDark();

const imgUrl = computed(() => isDark.value ? '/dev/dev-dark.png' : '/dev/dev.png')
</script>

<template>
  <div :class="$style.wrapper">
    <ProjectGrid />
    <div :class="$style.img_container">
      <img decoding="async" :src="imgUrl" />
    </div>
  </div>
</template>

<style lang="scss" module>
.wrapper {
  overflow: hidden;
  position: relative;
  width: 100%;
  margin-top: 30px;
  align-items: center;
  display: grid;
  grid-template-columns: repeat(1, minmax(0, 1fr));
  grid-template-rows: repeat(1, minmax(0, 1fr));

  .img_container {
    position: relative;
    justify-self: center;
    display: flex;
    align-items: center;
    width: 100%;
    grid-area: 1 / 1 / 1 / 1;
    max-width: 70%;
    transform: translateY(-1%);
    box-shadow:
      0 10px 15px -3px rgb(0 0 0 / 0.1),
      0 4px 6px -4px rgb(0 0 0 / 0.1);
    border-radius: 12px;
    height: 300px;

    & > img {
      max-width: 100%;
      height: auto;
      border-radius: 12px;
    }
  }
}

@media screen and (max-width: 890px) {
  .wrapper{
    .img_container{
      max-width: 80%;
    }
  }
}

@media screen and (max-width: 680px) {
  .wrapper{
    .img_container{
      height: fit-content;
    }
  }
}

@media screen and (max-width: 520px) {
  .wrapper{
    .img_container{
      max-width: 90%;
    }
  }
}

@media screen and (max-width: 420px) {
  .wrapper{
    margin-top: 50px;
  }
}
</style>
