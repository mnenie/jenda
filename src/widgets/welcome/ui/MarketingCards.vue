<script setup lang="ts">
import { computed } from 'vue';
import { useDark } from '@vueuse/core';
import { useI18n } from 'vue-i18n';
import { _cards } from '../config/cards';
import type { MarketingCard } from '../types';
import { UiCard } from '@/shared/ui';

const { tm } = useI18n();

const cards = computed(() => {
  const localeArr = tm('welcome.cards') as MarketingCard[];
  return _cards.map((card, i) => ({
    ...card,
    title: localeArr[i].title,
    description: localeArr[i].description
  }));
});

const colors = [
  { light: '#fff9f3', dark: '#2b2723' },
  { light: '#fbfbfc', dark: 'rgba(var(--zinc-rgb-600), 0.2)' },
  { light: '#f7f9fe', dark: '#232527' },
  { light: '#fef7f9', dark: '#302528' },
  { light: '#fbfbfc', dark: 'rgba(var(--zinc-rgb-600), 0.2)' }
];

const isDark = useDark();

const getCurrentColor = computed(() => {
  return (id: number) => {
    return !isDark.value ? colors[id].light : colors[id].dark;
  };
});

const currentWidth = computed(() => {
  return (width: string | undefined) => {
    return width ? width + 'px' : '100%';
  };
});
const currentHeight = computed(() => {
  return (height: string | undefined) => {
    return height ? height + 'px' : '100%';
  };
});
</script>

<template>
  <div :class="$style.wrapper">
    <div :class="$style.combinedCard">
      <UiCard
        v-for="card in cards.slice(0, 2)"
        :key="card.id"
        :class="$style.card"
        :style="{
          backgroundColor: getCurrentColor(card.id),
          maxWidth: currentWidth(card.width),
          height: currentHeight(card.height)
        }"
      >
        <h3 class="heading-3">{{ card.title }}</h3>
        <p class="text-base">{{ card.description }}</p>
        <img :src="isDark ? '/kanban-d.png' : '/kanban.png'" />
      </UiCard>
    </div>
    <div :class="$style.combinedCard">
      <UiCard
        v-for="card in cards.slice(2, 4)"
        :key="card.id"
        :class="$style.card"
        :style="{
          backgroundColor: getCurrentColor(card.id),
          maxWidth: currentWidth(card.width),
          height: currentHeight(card.height)
        }"
      >
        <h3 class="heading-3">{{ card.title }}</h3>
        <p class="text-base">{{ card.description }}</p>
        <img :src="isDark ? '/kanban-d.png' : '/kanban.png'" />
      </UiCard>
    </div>
    <div :class="$style.combinedCard">
      <UiCard
        :key="cards[4].id"
        :class="[$style.card, $style.flexGrowCard]"
        :style="{
          backgroundColor: getCurrentColor(cards[4].id),
          maxWidth: currentWidth(cards[4].width),
        }"
      >
        <h3 class="heading-3">{{ cards[4].title }}</h3>
        <p class="text-base">{{ cards[4].description }}</p>
        <img :src="isDark ? '/kanban-d.png' : '/kanban.png'" />
      </UiCard>
    </div>
  </div>
</template>

<style module lang="scss">
.wrapper {
  display: flex;
  margin: 70px 0 80px 0;
  width: 100%;
}

.combinedCard {
  gap: 20px;
  display: flex;
  flex-direction: column;
  flex-grow: 1;

  &:not(:last-child) {
    margin-right: 20px;
  }
}

.card {
  padding: 20px;
  position: relative;
  overflow: hidden;
  display: flex;
  flex-direction: column;
  width: 100%;
  flex-grow: 1;

  & > h3 {
    margin-bottom: 8px;
  }

  & > p {
    color: var(--zinc-500);
  }

  & > img {
    position: absolute;
    bottom: -260px;
    right: -360px;
    width: 720px;
    border-radius: 8px;
  }
}

:global(html.dark) {
  .card {
    & > p {
      color: var(--zinc-300);
    }
    & > img {
      border: 1px solid var(--zinc-700);
    }
  }
}
</style>
