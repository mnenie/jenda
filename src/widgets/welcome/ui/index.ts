import MarketingCards from './MarketingCards.vue';
import OfferJenda from './OfferJenda.vue';
import ProjectDemo from './ProjectDemo.vue';
import MarqueeList from './MarqueeList.vue';

import KanbanSection from './sections/KanbanSection.vue';
import CollaborativeSection from './sections/CollaborativeSection.vue';
import AboutSection from './sections/AboutSection.vue';
import ActivitySection from './sections/ActivitySection.vue';
import MembersSection from './sections/MembersSection.vue';
import ChatsSection from './sections/ChatsSection.vue';

export {
  AboutSection,
  MarketingCards,
  OfferJenda,
  ProjectDemo,
  MarqueeList,
  KanbanSection,
  CollaborativeSection,
  ActivitySection,
  MembersSection,
  ChatsSection
};
