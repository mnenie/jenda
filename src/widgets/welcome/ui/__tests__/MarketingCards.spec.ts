import { describe, it, expect } from 'vitest';
import { mount, VueWrapper } from '@vue/test-utils';
import '@/shared/lib/vitest-utils/cookiesI18n-mock';
import i18n from '@/shared/lib/i18n';
import MarketingCards from '../MarketingCards.vue';
import { nextTick, type ComponentPublicInstance } from 'vue';
import { UiCard } from '@/shared/ui';

// TODO: `ComponentPublicInstance` will need to set in other tests too
type MarketingCardsInstance = ComponentPublicInstance<
  {},
  {},
  {
    cards: any[];
    width: number;
    getCurrentColor: (id: number) => string;
    currentWidth: (width: string | undefined) => string;
    currentHeight: (height: string | undefined) => string;
  }
>;

describe('tests for MarketingCards.vue', () => {
  const wrapper = mount(MarketingCards, {
    global: {
      plugins: [i18n]
    }
  }) as VueWrapper<MarketingCardsInstance>;

  it('should be render correctly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render subcomponents', () => {
    expect(wrapper.findComponent(UiCard).exists()).toBe(true);
  });

  it('should count cards correctly', async () => {
    const cards = wrapper.vm.cards;
    wrapper.vm.width = 1900;
    await nextTick();
    expect(cards).toHaveLength(5);
  });

  it('returns correct width and height', () => {
    expect(wrapper.vm.currentWidth('500')).toBe('500px');
    expect(wrapper.vm.currentWidth(undefined)).toBe('100%');
    expect(wrapper.vm.currentHeight('100')).toBe('100px');
    expect(wrapper.vm.currentHeight(undefined)).toBe('100%');
  });

  it('renders card titles and descriptions correctly', () => {
    const _titles = wrapper.findAll('h3');
    const _descriptions = wrapper.findAll('p');

    expect(_titles.length).toBe(5);
    expect(_descriptions.length).toBe(5);

    _titles.forEach((title, index) => {
      expect(title.text()).toBe(wrapper.vm.cards[index].title);
    });
    _descriptions.forEach((desc, index) => {
      expect(desc.text()).toBe(wrapper.vm.cards[index].description);
    });
  });
});
