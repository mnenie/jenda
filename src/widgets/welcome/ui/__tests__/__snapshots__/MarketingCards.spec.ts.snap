// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`tests for MarketingCards.vue > should be render correctly 1`] = `
"<div class="wrapper">
  <div style="display: flex; gap: 20px;"><svg xmlns="http://www.w3.org/2000/svg" width="40" height="40" viewBox="0 0 24 24" fill="none" stroke="var(--zinc-600)" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide lucide-package-icon">
      <path d="m7.5 4.27 9 5.15"></path>
      <path d="M21 8a2 2 0 0 0-1-1.73l-7-4a2 2 0 0 0-2 0l-7 4A2 2 0 0 0 3 8v8a2 2 0 0 0 1 1.73l7 4a2 2 0 0 0 2 0l7-4A2 2 0 0 0 21 16Z"></path>
      <path d="m3.3 7 8.7 5 8.7-5"></path>
      <path d="M12 22V12"></path>
    </svg>
    <h2 style="font-size: 40px;" class="heading-2">All convenience in one place</h2>
  </div>
  <section>
    <div class="combinedCard">
      <div class="text-sm card card" style="background-color: rgb(255, 255, 255); max-width: 640px; height: 270px;">
        <h3 class="heading-3">Kanban 🧑‍💻</h3>
        <p class="text-base">Organize and visualize your projects with our intuitive Kanban board. Enhance productivity with ease.</p><img src="/dev-card.png" style="width: 720px; bottom: -260px; right: -360px;">
        <div class="user_container"></div>
      </div>
      <div class="text-sm card card" style="background-color: rgb(255, 255, 255); max-width: 640px; height: 300px;">
        <h3 class="heading-3">Collaborative 👥</h3>
        <p class="text-base">Improve your teamwork!</p><img src="/dev-card.png" style="width: 600px; bottom: -160px; right: 20px;">
        <div class="user_container"></div>
      </div>
    </div>
    <div class="combinedCard">
      <div class="text-sm card card" style="background-color: rgb(255, 255, 255); max-width: 500px; height: 420px;">
        <h3 class="heading-3">Members 🌐</h3>
        <p class="text-base">Invite new participants to your projects. Enjoy the development together!</p><img src="/dev-card.png" style="width: 600px; bottom: -230px; right: -290px;">
        <div class="user_container"></div>
      </div>
      <div class="text-sm card card" style="background-color: rgb(255, 255, 255); max-width: 500px; height: 100%;">
        <h3 class="heading-3">Activity 👔</h3>
        <p class="text-base">Analyze your activity and move forward.</p><img src="/dev-card.png" style="width: 700px; bottom: -200px; right: -500px;">
        <div class="user_container"></div>
      </div>
    </div>
    <div class="combinedCard">
      <div class="text-sm card card flexGrowCard" style="background-color: rgb(255, 255, 255); max-width: 100%;">
        <h3 class="heading-3">Chats 💬</h3>
        <p class="text-base">Communicate, share, discuss. Our convenience for you 😉</p><img src="/dev-card.png" style="width: 700px; bottom: 0px; right: -420px;">
        <div class="user_container"></div>
      </div>
    </div>
  </section>
</div>"
`;
